pkg/util/tls_config.go

```diff
@@ -81,7 +78,6 @@ func NewTLSConfigWithPassword(clientCert, clientKey, clientKeyPassword, caCert s
 		caCertPool := x509.NewCertPool()
 		caCertPool.AppendCertsFromPEM([]byte(caCert))
 		config.RootCAs = caCertPool
-		config.InsecureSkipVerify = skipVerify
```

Comment:
Should we make this configurable to end-users can opt-out in case they need to?

Reply:
I was thinking about the same.

Reply:
We could make it configurable, but I didn't because the user is supplying a trusted certificate here. I don't see why a user would supply a certificate at all if they don't want to verify the certificate chain.

Reply:
Make sense

Reply:
What if it's self-signed because it's a DEV environment? I think there are valid scenarios to make this configurable though.

Reply:
Even if we add it and the default is true; it's put to end-users to explicitly turn off and thus be aware that validation is not applied.

I would personally add this because I already know we'll get support questions on this :)

Reply:
And then there is also the fact that this is a breaking change for which we don't offer a way back.

Reply:
I agree with @tomkerkhove , we should provide a mechanism to disable the validation. Even though IMO we can introduce a breaking change this time (and we should indeed), we should keep the option of ignoring cert validation because we could face with users blocked due to this.
Is the CA used for authentication always the same as the CA used for TLS?

Reply:
When a user does not supply a CA in this section of the code, the operator uses the default for `InsecureSkipVerify`, which is of course `false`. There is no option to override that default. This PR only affects configurations that supply a set of root CAs to override the default root CAs from the OS. See https://pkg.go.dev/crypto/tls#Config:

```Go
// RootCAs defines the set of root certificate 
// that clients use when verifying server certificates.
// If RootCAs is nil, TLS uses the host's root CA set.
RootCAs *x509.CertPool
```

> What if it's self-signed because it's a DEV environment? I think there are valid scenarios to make this configurable though.

Will you explain how self-signed certificates are relevant? The user is configuring root CAs to trust. A self signed certificate still has a CA, so in that scenario, a user would supply the self signed cert's root CA to validate the self signed certificate without needing to disable certificate validation.

I don't think there is any use case where the user would supply trusted CAs and not want to use them to validate that the remote server's certificate chain is signed by one of the trusted CAs.

If it is necessary to make this configurable, I think it belongs at a higher level. It doesn't make sense to require a user to specify a trusted cert to be able to configure disabling certificate validation. When certificate validation is disabled,  root CAs are irrelevant.

> Is the CA used for authentication always the same as the CA used for TLS?

My understanding is that the CA is not used for authentication, it is only used to verify the remote server's certificate chain. When using TLS based authentication, the server only provides one cert chain to the client. Source: https://www.cloudflare.com/learning/access-management/what-is-mutual-tls/.

All that being said, I agree that this change could technically break the implicit contract that some users might have come to expect.

Reply:
If you are 100% sure then I'm fine